src/components/context-selection/PeriodSelector.js

```diff
@@ -1,20 +1,195 @@
-import { SelectorBarItem } from '@dhis2/ui'
+import i18n from '@dhis2/d2-i18n'
+import { generateFixedPeriods } from '@dhis2/multi-calendar-dates'
+import {
+    Button,
+    InputField,
+    SelectorBarItem,
+    SingleSelectField,
+    SingleSelectOption,
+} from '@dhis2/ui'
 import PropTypes from 'prop-types'
 import React, { useState } from 'react'
+import styles from './PeriodSelector.module.css'
 
-export const PeriodSelector = ({ selectedPeriodInfo }) => {
+const ANNUAL_REPORT_PERIOD_TYPES = [
+    { label: i18n.t('Yearly'), id: 'YEARLY' },
+    { label: i18n.t('Financial Year (April)'), id: 'FYAPR' },
+    { label: i18n.t('Financial Year (July)'), id: 'FYJUL' },
+    { label: i18n.t('Financial Year (October)'), id: 'FYOCT' },
+    // todo: wait, what? should this be Nov? see /api/periodTypes
+    { label: i18n.t('Financial Year (December)'), id: 'FYDEC' },
+]
+
+function PeriodTypeSelect({ periodType, setPeriodType, setPeriods }) {
+    return (
+        <SingleSelectField
+            label={i18n.t('Period type')}
+            placeholder={i18n.t('Choose a period type')}
+            selected={periodType}
+            onChange={({ selected }) => {
+                setPeriodType(selected)
+                // Reset period selection too
+                setPeriods([])
+            }}
+        >
+            {ANNUAL_REPORT_PERIOD_TYPES.map(({ label, id }) => (
+                <SingleSelectOption key={id} label={label} value={id} />
+            ))}
+        </SingleSelectField>
+    )
+}
+PeriodTypeSelect.propTypes = {
+    /** Expected to be an ISO ID of the period type */
+    periodType: PropTypes.string,
+    setPeriodType: PropTypes.func,
+    setPeriods: PropTypes.func,
+}
+
+/** Combines period and yearsForReference because their state is shared */
+const PeriodsSelect = ({
+    periodType,
+    periods,
+    setPeriods,
+    yearsForReference,
+    setYearsForReference,
+}) => {
+    const generatedPeriods = React.useMemo(() => {
+        return periodType
+            ? generateFixedPeriods({
+                  // todo: should this be systemInfo.calendar?
+                  calendar: 'gregory',
+                  periodType: periodType,
+                  year: new Date().getFullYear() - 1,
+                  // todo: internationalize; me.settings.keyUiLocale (or DB locale?)
+                  locale: 'en',
```

Comment:
Yeah. I asked that this be implemented with these as constants initially. I've added notes on the ticket for this as future internationalization work: https://dhis2.atlassian.net/browse/RWDQA-7

(I think we have used user's ui locale for dates where we've implemented this before)